src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/beanvalidation/BeanValidationQuickFix.java

```diff
@@ -1,82 +1,138 @@
 
 import com.intellij.psi.*;
 import com.intellij.psi.util.PsiTreeUtil;
+import io.openliberty.tools.intellij.lsp4jakarta.lsp4ij.JDTUtils;
 import io.openliberty.tools.intellij.lsp4jakarta.lsp4ij.Messages;
 import io.openliberty.tools.intellij.lsp4jakarta.lsp4ij.codeAction.proposal.ModifyModifiersProposal;
 import io.openliberty.tools.intellij.lsp4jakarta.lsp4ij.codeAction.proposal.RemoveAnnotationsProposal;
+import io.openliberty.tools.intellij.lsp4mp4ij.psi.core.java.codeaction.IJavaCodeActionParticipant;
 import io.openliberty.tools.intellij.lsp4mp4ij.psi.core.java.codeaction.JavaCodeActionContext;
+import io.openliberty.tools.intellij.lsp4mp4ij.psi.core.java.codeaction.JavaCodeActionResolveContext;
 import org.eclipse.lsp4j.CodeAction;
 import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.WorkspaceEdit;
 
 import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 /**
  * Quickfix for fixing {@link BeanValidationConstants#DIAGNOSTIC_CODE_STATIC} error by either action
  * 1. Removing constraint annotation on static field or method
  * 2. Removing static modifier from field or method
- * 
- * @author Leslie Dawson (lamminade)
  *
+ * @author Leslie Dawson (lamminade)
  */
-public class BeanValidationQuickFix {
+public class BeanValidationQuickFix implements IJavaCodeActionParticipant {
+
+    private static final Logger LOGGER = Logger.getLogger(BeanValidationQuickFix.class.getName());
+
+    @Override
+    public String getParticipantId() {
+        return BeanValidationQuickFix.class.getName();
+    }
 
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic) {
         List<CodeAction> codeActions = new ArrayList<>();
-        removeConstraintAnnotations(diagnostic, context.copy(), codeActions);
+        getRemoveConstraintAnnotationsCodeActions(diagnostic, context.copy(), codeActions);
 
         if (diagnostic.getCode().getLeft().equals(BeanValidationConstants.DIAGNOSTIC_CODE_STATIC)) {
-            removeStaticModifier(diagnostic, context.copy(), codeActions);
+            getRemoveStaticModifierCodeActions(diagnostic, context.copy(), codeActions);
         }
         return codeActions;
     }
 
-    private void removeConstraintAnnotations(Diagnostic diagnostic, JavaCodeActionContext context, List<CodeAction> codeActions) {
+    @Override
+    public CodeAction resolveCodeAction(JavaCodeActionResolveContext context) {
+
+        final CodeAction toResolve = context.getUnresolved();
+        List<Diagnostic> diagnostics = toResolve.getDiagnostics();
+
+
+        String message = toResolve.getTitle();
+
+        if (message.equals(Messages.getMessage("RemoveStaticModifier"))) {
+            resolveStaticModifierCodeAction(context);
+            return toResolve;
+        }
+
+        resolveRemoveConstraintAnnotationsCodeAction(context, diagnostics.size() > 0 ? diagnostics.get(0) : null);
```

Comment:
Related to my comment about retrieving the annotation name from `CodeActionResolveData`, I think you're only searching for the diagnostic here so you can compute the annotation name again. Looks like you could remove that parameter. This would also eliminate any risk of getting a `NullPointerException` since the reference is being used without checking whether it's `null` in the method.